charts/azure-k8s-metrics-adapter/templates/deployment.yaml

```diff
@@ -44,7 +44,7 @@ spec:
               containerPort: {{ .Values.adapterSecurePort }}
               protocol: TCP
           env:
-          {{- if not (eq "msi" .Values.azureAuthentication.method) }}
+          {{- if and (not (eq "msi" .Values.azureAuthentication.method)) (not (eq "aadPodIdentity" .Values.azureAuthentication.method)) }}
```

Comment:
Would this be easier to understand if we used ` if eq "clientSecret" .Values.azureAuthentication.method` instead of the double negative?

I think the tenant id and secret should be deployed together.  Is there a scenario where they could be deployed independently?

Reply:
Hmm, you don't need the tenant id when using client certificate authentication?

Reply:
Yes you are right.  I think this would be more readble if used the postive instead of double negative. So if clientsecret or client cert then deploy otherwise leave out. What do you think?  

I plan on removing support for msi now that pod identity is supported.

Reply:
Yep, makes sense. I've done the update :)